### ISSUE TITLE
Incorrect Behavior of Quantification Operators

### ISSUE THREAD

**Issue by quentinchambers** [tree-sitter/tree-sitter-graph#120]
It seems that the quantification operator behaves incorrectly for the following query:

`function_definition parameters:(parameters (identifier)* @name) @parameters)@func`

My expectation is that @name would contain a list of identifiers but it only seems to contain the first identifier. My test is done using python with the following source

`
def start(a, b):
	x=a
`

The query correctly grabs each identifier without the quantification operator. I've also done a similar query using `(function_definition body:(block  (expression_statement)* @expr_stmts)@block)@func` which behaves as expected. Since the query matches the function parameters correctly, I assume this is an issue with the graph library. I was using version 0.7 for this test.

**Comment by hendrikvanantwerpen** [comment]
Thank you for reporting this! I'll try to reproduce this. It's the small Python snippet enough to trigger the behavior?

This could also be a bug in the underlying Tree-sitter query engine we use, since TSG doesn't do much with those queries before passing them on.

**Comment by quentinchambers** [comment]
Yea I reduced my test to just that snippet and was experiencing the issue. The code formatting is off though so `x=a` should be on a newline and indented. This seemed to be an issue with using the quantifier to grab function parameters. 

Oh I didn't realize those quantifiers existed for tree-sitter queries. I just ran a test using the query directly and it has the same problem. Should I open a new issue with the tree-sitter repo? 

I'm also pretty new to tree-sitter so I hope this is an actual issue and not just a misunderstanding of mine!
